preview/Win7Msix/Win7MSIXInstallerLib/inc/IPackageManager.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <functional>
+#include "IMsixResponse.hpp"
+#include "IPackage.hpp"
+#include "DeploymentOptions.hpp"
+
+
+namespace Win7MsixInstallerLib {
+    class IPackageManager
+    {
+    public:
+        virtual IMsixResponse * AddPackageAsync(const std::wstring & packageFilePath, DeploymentOptions options) = 0;

```

Comment:
>AddPackageAsync [](start = 32, length = 15)

i think this might make more sense to take in the callback function, rather than expose SetCallback on the IMsixResponse interface; internally, we'd want to set the callback asap before returning the imsixresponse object because the addpackage will begin processing as the imsixresponse object is returned. 

Also, we need to make the msixresponse pointer a shared pointer so that it doesn't get deleted when msixrequest finishes.

Reply:
What do you prefer:

```
IMsixResponse * response = CreateMsixResponse();
response.SetCallback(...);
HResult res = AddPackageAsync(packageFilePath, options, &response)
```

or 

```
HResult res = AddPackageAsync(packageFilePath, options, [](){ ... })
```

?

Reply:
I think IMsixResponse * response = AddPackageAsync(packageFilePath, options, [](){...}); would be best. Not sure if you meant for the second of your options return value to change to HRESULT because now the response object is nowhere to be found?

Reply:
Faulty copy/paste. I will modify that asap

Reply:
@wcheng-msft, all pointers exposed by the library are now shared_ptr/unique_ptr + callback added